File: solutions_5639104758808576_0/Python/Davidinosauro/A_standing_ovation.py
#!/usr/bin/env python2
from __future__ import division, print_function

import sys
import math
from itertools import izip, tee
from pprint import pprint as pp
from collections import Counter
import operator as op

def dbg(*args, **kwargs):
    return
    kwargs['file'] = sys.stderr
    print(*args, **kwargs)


def result(N, s):
    standing = 0
    friends = 0
    dbg(s)
    for shiness, count in enumerate(s):
        if standing < shiness:
            friends += (shiness - standing)
            standing += (shiness - standing)
        standing += count
        
    return friends


if __name__ == '__main__':
    #sys.setrecursionlimit(max(2000, sys.getrecursionlimit()))
    T = int(sys.stdin.readline().strip())
    for t in range(T):
        print('===', t+1, '===', file=sys.stderr)

        N, s = tuple(tk for tk in sys.stdin.readline().strip().split())
        N = int(N)
        s = tuple(int(c) for c in s)

        res = result(N, s)
        print('Case #{}: {}'.format(t+1, res))
        sys.stdout.flush()

File: solutions_5639104758808576_1/C++/btzy/A-standing-ovation.cpp
#include<cstdio>
using namespace std;
int tc,s,standing,p,friends;
char cp;
int main(){
    scanf("%d",&tc);
    for(int ct=1;ct<=tc;++ct){
        scanf("%d",&s);
        standing=0;
        friends=0;
        for(int i=0;i<=s;++i){
            scanf(" %c",&cp);
            p=cp-'0';
            while(p>0){
                while(p>0&&standing>=i){
                    --p;
                    ++standing;
                }
                if(p>0){
                    ++friends;
                    ++standing;
                }
            }
        }
        printf("Case #%d: %d\n",ct,friends);
    }
}
